src/decoders/reverse_decoder.rs

```diff
@@ -8,29 +8,22 @@ use super::interface::Decoder;
 
 use log::trace;
 
-/// .decoder is never used, so Rust considers this dead code
-/// Really it's just a co-reference to the Decoder in `interface.rs`
-#[allow(dead_code)]
-pub struct ReverseDecoder {
-    decoder: Decoder,
-}
+pub struct ReverseDecoder;
 
 impl ReverseDecoder {
-    pub fn new() -> Self {
-        Self {
-            decoder: Decoder {
-                name: "Reverse",
-                description: "Reverses a string. stac -> cats",
-                link: "http://string-functions.com/reverse.aspx",
-                tags: vec!["reverse", "decoder"],
-                expected_runtime: 0.01,
-                expected_success: 1.0,
-                failure_runtime: 0.01,
-                normalised_entropy: vec![1.0, 10.0],
-                // I have never seen a reversed string in a CTF
-                // or otherwise
-                popularity: 0.2,
-            },
+    pub fn new() -> Decoder<ReverseDecoder> {
+        Decoder {
+            decoder: ReverseDecoder,
```

Comment:
How do does this work? How do we get the name of the decoder as a string?

Reply:
i thought name was required just to identify the decoder and that can also be done by struct. If we want name, we can just add a `name` field as before.